src/_P036_FrameOLED.ino

```diff
@@ -718,10 +734,26 @@ boolean Plugin_036(byte function, struct EventStruct *event, String& string)
             // content functions
             success = true;
             String NewContent = parseStringKeepCase(string, 3);
-            P036_DisplayLinesV1[LineNo-1].Content = NewContent;
+            if (!safe_strncpy(P036_DisplayLinesV1[LineNo-1].Content, NewContent, P36_NcharsV1)) {
+              addHtmlError(getCustomTaskSettingsError(LineNo-1));
+            }
+            display->setFont(ScrollingPages.Font);
+            P036_DisplayLinesV1[LineNo-1].Content[P36_NcharsV1-1] = 0; // Terminate in case of uninitalized data
             nextFrameToDisplay = LineNo / ScrollingPages.linesPerFrame; // next frame shows the new content
 
-//            int LineMode = event->Par3; // not implemented yet
+            P036_DisplayLinesV1[LineNo-1].reserved = (event->Par3 & 0xFF); // not implemented yet
+
+            // calculate Pix length of new Content
+            uint16_t PixLength = display->getStringWidth(String(P036_DisplayLinesV1[LineNo-1].Content));
+            if (PixLength > 255) {
+              addHtmlError(String(F("Pixel length of ")) + String(PixLength) + String(F(" too long for line! Max. 255 pix!")));
```

Comment:
You cannot know the length here.

The content should also be processed by `P36_parseTemplate`.
Otherwise some characters will not be shown using the correct font entry.
Also you want to make sure system variables or references to other plugin values are processed. That's also done in the `P36_parseTemplate` function.
But that also makes the actual string length variable.

Reply:
This is in PLUGIN_WRITE (sending a command from the command line!).
The command is processed by parseString() and event->Par
The StringWidth can be calculated here:
Why not, a few lines above the correct font is set (display->setFont(ScrollingPages.Font);)

Reply:
The `P36_parseTemplate` function does also replace some characters to match them with the OLED font.
So if they are not replaced, the width estimate will be wrong as the string may contain characters not present in the font.

Reply:
I see, already going to change to `P36_parseTemplate` function.